### PR TITLE
fix next mapping

### PR DESCRIPTION
**Pull request by bend-n**

### CODE REVIEW

File: src/main/PlagueMain.java [hunk 1]
```diff
@@ -376,64 +370,68 @@ public void init() {
...
+                event.tile.build.items
+                        .add(isSerpulo ? PlagueData.survivorIncrementSerpulo : PlagueData.survivorIncrementErekir);
+                return;
+            }
 
-                }
+            for (ItemStack stack : isSerpulo ? PlagueData.survivorLoadoutSerpulo : PlagueData.survivorLoadoutErekir) {
+                Call.setItem(event.tile.build, stack.item, stack.amount);
             }
         });
```
Comment (nilederg) [inline src/main/PlagueMain.java:436]: Thanks for incorporating my work

File: src/main/PlagueMain.java [hunk 2]
```diff
@@ -721,13 +730,45 @@ public void registerClientCommands(CommandHandler handler) {
...
+                if (
+                // nothing there
+                check == null
+                        // deep water
+                        || check.floor().isDeep()
+                        || (type == check.block() && check.build != null
+                                && type.rotate)
+                        || // same block, same rotation
+                        !check.floor().placeableOn // solid wall
+                )
```
Comment (nilederg) [inline src/main/PlagueMain.java]: Ugly logic - please fix

File: src/main/PlagueMain.java [hunk 3]
```diff
@@ -1084,38 +1128,29 @@ void loadMap(String[] args) {
...
         PlagueGenerator.defaultOres(world.tiles, isSerpulo);
 
         world.endMapLoad();
-
-        rules.enemyCoreBuildRadius = 75 * 7;
-        rules.hiddenBuildItems = map.rules().hiddenBuildItems;
+        rules.hiddenBuildItems = (isSerpulo ? Items.erekirOnlyItems : PlagueData.serpuloOnlyItems).asSet();
+        rules.bannedBlocks = isSerpulo ? ObjectSet.with() : ObjectSet.with(Blocks.plastaniumConveyor);
         rules.hideBannedBlocks = true;
+
```
Comment (nilederg) [inline src/main/PlagueMain.java:1154]: Since this isn't part of the diffs, I'll allow it despite having no knowledge of the firstRun boolean

**Review by nilederg — CHANGES_REQUESTED** [review]
Clean up ugly logic, please

**Comment by bend-n** [comment]
ugly logic fixed

File: src/main/PlagueMain.java [hunk 4]
```diff
@@ -721,13 +730,46 @@ public void registerClientCommands(CommandHandler handler) {
...
+                if (
+                    // nothing there
+                    check == null
+                    // deep water
+                    || check.floor().isDeep()
+                    // same block, same rotation
+                    || (type == check.block() && check.build != null && type.rotate)
+                    // solid wall
+                    || !check.floor().placeableOn
+                )
```
Comment (nilederg) [inline src/main/PlagueMain.java:763]: Thanks, this is readable now

**Review by nilederg — APPROVED** [review]
Looks good to me, all code is readable, changes are useful.

**Pull request merged**